src/components/common/needsShutdown.jsx

```diff
@@ -0,0 +1,161 @@
+        getIfaceSourceName(inactiveIface) !== getIfaceSourceName(iface);
+}
+
+export function needsShutdownVcpu(vm) {
+    return ((vm.vcpus.count !== vm.inactiveXML.vcpus.count) ||
+            (vm.vcpus.max !== vm.inactiveXML.vcpus.max) ||
+            (vm.cpu.sockets !== vm.inactiveXML.cpu.sockets) ||
+            (vm.cpu.threads !== vm.inactiveXML.cpu.threads) ||
+            (vm.cpu.cores !== vm.inactiveXML.cpu.cores));
+}
+
+export function needsShutdownCpuModel(vm) {
+    /* The live xml shows what host-model expanded to when started
+     * This is important since the expansion varies depending on the host and so needs to be tracked across migration
+     */
+    if (vm.inactiveXML.cpu.mode === 'host-model')
+        return !(vm.cpu.mode == 'host-model' || vm.cpu.model === vm.capabilities.cpuHostModel);
+    if (vm.inactiveXML.cpu.mode === 'host-passthrough')
+        return vm.cpu.mode !== 'host-passthrough';
+    if (vm.inactiveXML.cpu.mode === 'custom')
+        return vm.cpu.mode !== 'custom' || vm.cpu.model !== vm.inactiveXML.cpu.model;
+
+    return false;
+}
+
+export function needsShutdownWatchdog(vm) {
+    return vm.persistent && vm.state === "running" && vm.inactiveXML.watchdog.action !== vm.watchdog.action;
+}
+
+export function getDevicesRequiringShutdown(vm) {
+    if (!vm.persistent)
+        return [];
+
+    const devices = [];
+
+    // DISKS
+    for (const target in vm.disks) {
+        if (needsShutdownDiskAccess(vm, target)) {
+            devices.push(_("Disk"));
+            break;
+        }
+    }
+
+    // INTERFACES
+    for (const iface of vm.interfaces) {
+        if (needsShutdownIfaceType(vm, iface) ||
+            needsShutdownIfaceModel(vm, iface) ||
+            needsShutdownIfaceSource(vm, iface)) {
+            devices.push(_("Network interface"));
+            break;
+        }
+    }
+
+    // VCPU
+    if (needsShutdownVcpu(vm))
+        devices.push(_("vCPUs"));
+
+    // CPU
+    if (needsShutdownCpuModel(vm))
+        devices.push(_("CPU"));
+
+    // Watchdog
+    if (needsShutdownWatchdog(vm))
+        devices.push(_("Watchdog"));
+
+    return devices;
+}
+
+export const NeedsShutdownTooltip = ({ iconId, tooltipId }) => {
+    return (
+        <Tooltip id={tooltipId} content={NEEDS_SHUTDOWN_MESSAGE}>
+            <PendingIcon color="orange" id={iconId} />
+        </Tooltip>
+    );
+};
+
+export const NeedsShutdownAlert = ({ idPrefix }) =>
+    <Alert isInline variant='warning' id={`${idPrefix}-idle-message`} title={NEEDS_SHUTDOWN_MESSAGE} />;
+
+export const VmNeedsShutdown = ({ vm }) => {
+    const devices = getDevicesRequiringShutdown(vm);
+
+    if (devices.length === 0)
+        return;
+
+    const body = (
+        <>
+            <List isPlain isBordered>
+                { devices.map(device => {
+                    return (
+                        <ListItem key={device}>
+                            {cockpit.format(_("$0 configuration was changed"), device)}
```

Comment:
This is missing a period in the end, as the next sentence has one.

Also when more elements are there I find it very repetitive. Maybe let's remove the list, bring up the other string from the footer into the body and write something like:
```
<TextContent>
<Text>_("Changes to the domain's configuration only take effect after a complete shutdown and restart of the domain.")}
</Text>
<Text>cockpit.format(_("The are  pending changes for $0."), devices.join(", ")) </Text>
</TextContent>
```

Reply:
Hey. That was initial idea, but the issue is that in https://github.com/cockpit-project/cockpit-machines/pull/986 I was to add an additional message, like "Watchdog configuration was changed, but the VM needs shutdown. Watchdog live edit was not possible: Not enough PCI slots available".

So @garrett suggested the list: https://github.com/cockpit-project/cockpit-machines/pull/986#pullrequestreview-1362435505 , and we can render additional information for each device (each list item).

Reply:
Agreed.

I wonder if we should make this a list of configuration items are out of sync?

"Some configuration changes only take effect after a fresh boot:"

And then a list.

Reply:
@garrett Something like this:
![Screenshot from 2023-04-03 13-08-55](https://user-images.githubusercontent.com/42733240/229492690-877c229e-f954-43b3-b176-096c46a4d095.png)

Reply:
Hmm, there should probably be some padding between the list and the sentence above, updated it to this:
![Screenshot from 2023-04-03 13-14-29](https://user-images.githubusercontent.com/42733240/229493895-a70e0030-d8cb-4e16-b5d0-d917d79c0e14.png)

Reply:
Something like this:
![Screenshot from 2023-04-03 13-17-55](https://user-images.githubusercontent.com/42733240/229494463-f4a6d1e3-ab06-49a1-8b94-ca3ed3e307c6.png)

Reply:
We're already using blue for the action and "running" on the list. Perhaps it should be some other color.

Red: Error
Yellow/Orange: Warning
Green: Everything's good
Blue: Action
Grey: Actionless (stopped, etc.)

What's left?

https://www.patternfly.org/v4/guidelines/colors/#color-palette

Purple? Or reusing grey? (There's also cyan, but that's still blueish/greenish. And there's light green too. Purple might be too close to blue however.)

Reply:
But, yeah, that's looking more correct.

We could even add a

"[Reboot now] Cancel" to the bottom and treat it like a specifically-placed dialog. (Cancel just closing it like the X). This would give someone more guidance as what to do.

Additionally, we need to change the rebooting action everywhere to do a shutdown and restart if the configuration state is dirty like this. It's basically the same thing from the user's point of view, but we'd handle the details to make sure the machine is put into the expected state.